assets/app/scripts/controllers/create/createFromImage.js

```diff
@@ -100,6 +100,11 @@ angular.module("openshiftConsole")
             scope.buildConfig.contextDir = annotations.sampleContextDir || "";
           };
 
+          scope.usingSampleRepo = function() {
+            var sampleRepo = _.get('scope.image.metadata.annotations.sampleRepo');
+            return sampleRepo && scope.buildConfig.sourceUrl === sampleRepo;
```

Comment:
Never null/undefined. `scope.buildConfig` is not the actual bc, it's an object used to bind the form controls, and `sourceUrl` is a required field in this page. Removed the `sampleRepo &&` part.